services/databaseEncryptionService.ts

```diff
@@ -0,0 +1,163 @@
+import * as base64 from 'base64-js';
+
+const SECURE_STORE_KEYS = {
+  ENCRYPTION_KEY: 'encryption_key',
+};
+
+const ASYNC_STORAGE_KEYS = {
+  ENCRYPTION_MODE: 'encryption_mode',
+};
+
+export type EncryptionMode = 'basic' | 'protected';
+
+export class EncryptionError extends Error {
+  constructor(public code: keyof typeof ERROR_CODES, message: string) {
+    super(message);
+    this.name = 'EncryptionError';
+  }
+}
+
+export const ERROR_CODES = {
+  USER_CANCELLED: 'USER_CANCELLED',
+  KEY_NOT_FOUND: 'KEY_NOT_FOUND',
+} as const;
+
+let keyCache: Uint8Array | null = null;
+let initializationPromise: Promise<void> | null = null;
+
+function generateRandomKey(): Uint8Array {
+  return Crypto.getRandomBytes(32);
+}
+
+function isCancellationError(error: unknown): boolean {
+  if (!(error instanceof Error)) return false;
+  
+  const name = (error as any).name;
+  if (name === 'NotAllowedError' || name === 'ERR_CANCELED') return true;
+  
+  const code = (error as any).code;
+  if (code === 'ERR_CANCELED' || code === 'USER_CANCELLED') return true;
+  
+  return error.message.includes('cancel');
+}
+
+function classifyError(error: unknown): never {
+  if (error instanceof EncryptionError) {
+    throw error;
+  }
+
+  if (isCancellationError(error)) {
+    throw new EncryptionError(ERROR_CODES.USER_CANCELLED, 'Authentication was cancelled');
+  }
+
+  throw error;
+}
+
+async function createNewKey(): Promise<Uint8Array> {
+  const key = generateRandomKey();
+  await SecureStore.setItemAsync(SECURE_STORE_KEYS.ENCRYPTION_KEY, base64.fromByteArray(key), { requireAuthentication: false });
+  await setStoredEncryptionMode('basic');
+  return key;
+}
+
+async function loadExistingKey(): Promise<Uint8Array> {
+  const mode = await getStoredEncryptionMode();
+  const keyBase64 = await SecureStore.getItemAsync(SECURE_STORE_KEYS.ENCRYPTION_KEY, { requireAuthentication: mode === 'protected' });
+  
+  if (!keyBase64) {
+    throw new EncryptionError(ERROR_CODES.KEY_NOT_FOUND, 'Encryption key is missing. Your data cannot be decrypted.');
+  }
+
+  return base64.toByteArray(keyBase64);
+}
+
+export async function initializeEncryption(): Promise<void> {
+  if (keyCache) {
+    return;
+  }
+
+  if (initializationPromise) {
+    return initializationPromise;
+  }
+
+  initializationPromise = (async () => {
+    try {
+      keyCache = await loadExistingKey();
+    } catch (error) {
+      if (error instanceof EncryptionError && error.code === ERROR_CODES.KEY_NOT_FOUND) {
+        keyCache = await createNewKey();
+      } else {
+        throw classifyError(error);
+      }
+    } finally {
+      initializationPromise = null;
+    }
+  })();
+
+  return initializationPromise;
+}
+
+async function getStoredEncryptionMode(): Promise<EncryptionMode> {
+  try {
+    const mode = await AsyncStorage.getItem(ASYNC_STORAGE_KEYS.ENCRYPTION_MODE);
+    return mode === 'protected' ? 'protected' : 'basic';
+  } catch {
+    return 'basic';
+  }
```

Comment:
```suggestion
  const mode await AsyncStorage.getItem(ASYNC_STORAGE_KEYS.ENCRYPTION_MODE);
  return mode === 'protected' ? 'protected' : 'basic';
```

Correct me if I'm wrong, but `AsyncStorage.getItem()` would return `null` if the item is missing and it'd throw an error when something else goes wrong - in which case you don't want to silence it!